### ISSUE TITLE
Larger binaries produced by zig-linux-x86_64-0.14.0-dev.2414+ba37a4369

### ISSUE THREAD

**Issue by fwsGonzo** [ziglang/zig#22192]
### Zig Version

0.14.0-dev.2414+ba37a4369

### Steps to Reproduce and Observed Behavior

It looks like a change has made binaries much bigger from one nightly to the next.

> zig-linux-x86_64-0.14.0-dev.2384+cbc05e0b1
```
$ ls -lah bin
total 4,5M
drwxrwxr-x 2 gonzo gonzo 4,0K des.   8 18:50 .
drwxrwxr-x 7 gonzo gonzo 4,0K des.   8 18:50 ..
-rwxrwxr-x 1 gonzo gonzo 589K des.   8 18:49 asm
-rwxrwxr-x 1 gonzo gonzo 615K des.   8 18:50 cjit
-rwxrwxr-x 1 gonzo gonzo 231K des.   8 18:49 hello_world
-rwxrwxr-x 1 gonzo gonzo 762K des.   8 18:49 luajit
-rwxrwxr-x 1 gonzo gonzo 875K des.   8 18:49 mirjit
-rwxrwxr-x 1 gonzo gonzo 1,4M des.   8 18:50 robust_weight_transfer
```

> zig-linux-x86_64-0.14.0-dev.2414+ba37a4369
```sh
$ ls -lah bin
total 13M
drwxrwxr-x 2 gonzo gonzo 4,0K des.   9 07:36 .
drwxrwxr-x 7 gonzo gonzo 4,0K des.   9 07:36 ..
-rwxrwxr-x 1 gonzo gonzo 739K des.   9 07:36 asm
-rwxrwxr-x 1 gonzo gonzo 1,5M des.   9 07:36 cjit
-rwxrwxr-x 1 gonzo gonzo 398K des.   9 07:36 hello_world
-rwxrwxr-x 1 gonzo gonzo 929K des.   9 07:36 luajit
-rwxrwxr-x 1 gonzo gonzo 1,0M des.   9 07:36 mirjit
-rwxrwxr-x 1 gonzo gonzo 8,5M des.   9 07:36 robust_weight_transfer
```

I haven't changed anything in my build systems, other than updating Zig. Could this be related to https://github.com/ziglang/zig/pull/22167 somehow?

### Expected Behavior

Same incredibly small build sizes as before.

**Comment by Rexicon226** [comment]
There are no steps to reproduce your bug report, just the observed behaviour. What's the easiest way you've found to reproduce this size increase?

**Comment by Rexicon226** [comment]
How interesting 🤔. This seems to be because of the PR you mentioned.
cc @alexrp

**Comment by fwsGonzo** [comment]
Right, but I don't think that PR broke anything anymore. I just think I discovered a way to reduce the sizes. The programs all seem to be working fine. Applying `-flto` I could reduce them even further. It's really quite something.

But I don't know what the way forward is with this. It's strange that pulling in fake `atomic` reduced a single program from 8.5MB down to 1.4MB.

**Comment by alexrp** [comment]
Can you show the compile and link command lines that CMake generates, both before and after the issue started happening?
